controllers/eventsource/resource.go

```diff
@@ -240,42 +252,25 @@ func buildDeployment(args *AdaptorArgs, eventBus *eventbusv1alpha1.EventBus) (*a
 				},
 			},
 		})
-		volumeMounts = append(volumeMounts, corev1.VolumeMount{Name: "auth-volume", MountPath: common.EventBusAuthFileMountPath})
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      "auth-volume",
+			MountPath: common.EventBusAuthFileMountPath,
+		})
 	}
-	deploymentSpec.Template.Spec.Volumes = volumes
-	deploymentSpec.Template.Spec.Containers[0].VolumeMounts = volumeMounts
 
-	envs := deploymentSpec.Template.Spec.Containers[0].Env
-	envs = append(envs, envVars...)
-	deploymentSpec.Template.Spec.Containers[0].Env = envs
+	// secrets
+	volSecrets, volSecretMounts := common.VolumesFromSecretsOrConfigMaps(common.SecretKeySelectorType, secretObjs...)
+	volumes = append(volumes, volSecrets...)
+	volumeMounts = append(volumeMounts, volSecretMounts...)
 
-	vols := []corev1.Volume{}
-	volMounts := []corev1.VolumeMount{}
-	oldVols := deploymentSpec.Template.Spec.Volumes
-	oldVolMounts := deploymentSpec.Template.Spec.Containers[0].VolumeMounts
-	if len(oldVols) > 0 {
-		vols = append(vols, oldVols...)
-	}
-	if len(oldVolMounts) > 0 {
-		volMounts = append(volMounts, oldVolMounts...)
-	}
-	volSecrets, volSecretMounts := common.VolumesFromSecretsOrConfigMaps(eventSourceCopy, common.SecretKeySelectorType)
-	if len(volSecrets) > 0 {
-		vols = append(vols, volSecrets...)
-	}
-	if len(volSecretMounts) > 0 {
-		volMounts = append(volMounts, volSecretMounts...)
-	}
-	volConfigMaps, volCofigMapMounts := common.VolumesFromSecretsOrConfigMaps(eventSourceCopy, common.ConfigMapKeySelectorType)
-	if len(volConfigMaps) > 0 {
-		vols = append(vols, volConfigMaps...)
-	}
-	if len(volCofigMapMounts) > 0 {
-		volMounts = append(volMounts, volCofigMapMounts...)
-	}
+	// config maps
+	volConfigMaps, volCofigMapMounts := common.VolumesFromSecretsOrConfigMaps(common.ConfigMapKeySelectorType, eventSourceCopy)
+	volumeMounts = append(volumeMounts, volCofigMapMounts...)
+	volumes = append(volumes, volConfigMaps...)
 
-	deploymentSpec.Template.Spec.Volumes = vols
-	deploymentSpec.Template.Spec.Containers[0].VolumeMounts = volMounts
+	deploymentSpec.Template.Spec.Containers[0].Env = append(deploymentSpec.Template.Spec.Containers[0].Env, env...)
+	deploymentSpec.Template.Spec.Containers[0].VolumeMounts = append(deploymentSpec.Template.Spec.Containers[0].VolumeMounts, volumeMounts...)
+	deploymentSpec.Template.Spec.Volumes = append(deploymentSpec.Template.Spec.Volumes, volumes...)
```

Comment:
Regarding risk to Jetstream/nats, this code is generic to all: do you think there should be any additional testing done for this or do you think it's been covered?

Reply:
(same with controllers/sensor/resource.go below)

Reply:
I can add some more tests! I think that's a good idea